stl/inc/xstring

```diff
@@ -4269,55 +4298,62 @@ template <class _Elem, class _Traits, class _Alloc>
 _NODISCARD basic_string<_Elem, _Traits, _Alloc> operator+(
     const basic_string<_Elem, _Traits, _Alloc>& _Left, const basic_string<_Elem, _Traits, _Alloc>& _Right) {
     // return string + string
-    basic_string<_Elem, _Traits, _Alloc> _Ans;
-    _Ans.reserve(_Left.size() + _Right.size());
-    _Ans += _Left;
-    _Ans += _Right;
-    return _Ans;
+    if (_Left.max_size() - _Left.size() < _Right.size()) {

```

Comment:
Debug codegen question for all operators: Should you factor out the calls to each `.size()` since they are repeated in both the overflow check and the actual construction?

Reply:
Who are you and what did you do with Stephan?